### ISSUE TITLE
Do not use --ignore-platform-reqs

### ISSUE THREAD

**Issue by dnsl48** [silverstripe/cow#118]
Reported by @dhensby at #102 

## Description

We should never use --ignore-platform-reqs with composer. Whilst it appeared to be helpful at the time it's actually making things worse.

Cow needs to define the platform using composer config rather than outright ignoring the platform (which is effectively an --install-highest flag).

The killer problem with --ignore-platform-reqs is that even if we use composer config to define the platform (hint: we do) the flag means composer ignores that as well as our actual platform constraints.

All in all, it results in creating archives which only run on the highest of all our valid dependencies (PHP 7.2) rather than the lowest (5.6.0)

tl;dr --ignore-platform-reqs is evil, we must destroy it

**Comment by tractorcow** [comment]
Duplicate of https://github.com/silverstripe/cow/issues/80 just in case you missed it.  :)
